packages/itwin/grouping-mapping-widget/src/widget/components/CustomCalculationTable.tsx

```diff
@@ -6,30 +6,50 @@ import {
   SvgDelete,
```

Comment:
This component's only purpose is to render the data and handle callbacks. Would it be possible to do this differently? Maybe wrap this component with the validation logic. Intercept the callbacks and take advantage of them.

Reply:
I agree to wrap the component with the validation logic cause most of the component is currently just the validation logic. But I was hesitant to completely restructure this component as I saw a structure being used for the other tables (GroupPropertyTable and CalculatedPropertyTable) and thus have left it out for now. Would love to discuss the suggested changes further !!